src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -241,7 +241,7 @@ protected override async Task CreateVersionsAsync(CancellationToken cancellation
             // first add all the available versions to be updated
             foreach (var version in allVersionsAllowed)
             {
-                _versions.Add(new DisplayVersion(version.version, string.Empty, isDeprecated: version.isDeprecated));
```

Comment:
in this you switched DisplayVersion 2nd argument from empty to null, but in PackageDetailControlModel on line 175 you did the opposite: https://github.com/NuGet/NuGet.Client/pull/4687/files#diff-0b6aa73227964074dc06409994723b4f7d158fe4edc669b5c8229c1c3d235bd4R175. 

It seems like null and empty are considered equivalent, so maybe we don't need this file change at all.

Reply:
I use this for a little trick in the filter. https://github.com/NuGet/NuGet.Client/blob/2c115bdb929f6589e22ab17c28ac1d15467433c6/src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs#L246. 

The combobox looks like this and example:

Version Installed           6.0.0
Latest Prerelease          Latest stable 6.0.0-beta
Latest Stable                 
null separator               null separator
All Versions                  6.0.0-beta
                                     6.0.0
                                     6.1.1

If the user types 6.* I want to show only versions below the null separator to avoid duplicating items. For this if I add `string.Empty` to the Version Installed and leave `null` to the rest of versions I can filter out by `AddiontionalInfo` since everything up from the null separator will have a Latest stable/prerelease string or a `string.Empty` that will not affect the UI. With this I also eliminated the `IsBeforeNullSeparator` property that was weird

Reply:
Ok, I get it now. 

It's pretty hard to figure out :D